jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/BufferUtil.java

```diff
@@ -595,6 +595,21 @@ public static void readFrom(Path path, ByteBuffer buffer) throws IOException
         }
     }
 
+    public static void readFrom(ReadableByteChannel newReadableByteChannel, long contentLengthValue, ByteBuffer byteBuffer) throws IOException
+    {
+        int pos = BufferUtil.flipToFill(byteBuffer);
+        try (ReadableByteChannel channel = newReadableByteChannel)
+        {
+            int read = 0;
+            while (read != contentLengthValue)
+                read += channel.read(byteBuffer);
+        }
+        finally
+        {
+            BufferUtil.flipToFlush(byteBuffer, pos);
```

Comment:
Why flipping in a `finally` block?

Reply:
I think Greg suggested this in previous review because `channel.read` can throw.

Reply:
Thinking about it, it doesn't matter either way. If the method throws, the buffer cannot be trusted.